package.json

```diff
@@ -72,13 +73,23 @@
     "test": "react-scripts test",
     "eject": "react-scripts eject",
     "lint-js": "eslint 'src/**/*.{js,jsx}'",
-    "lint-ts": "tslint -c tslint.json --project ./tsconfig.json --format stylish 'src/**/*.{ts,tsx}'"
+    "lint-ts": "tslint -c tslint.json --project ./tsconfig.json --format stylish 'src/**/*.{ts,tsx}'",
+    "test:coverage": "set CI=true && react-scripts test --coverage",
+    "coveralls": "set CI=true && yarn test --runInBand --verbose --coverage=true --forceExit --detectOpenHandles && cat ./coverage/lcov.info | coveralls"
   },
   "jest": {
     "collectCoverageFrom": [
       "src/**/*.{ts,tsx}",
       "!src/index.tsx"
-    ]
+    ],
+    "coverageThreshold": {
+      "global": {
+        "branches": 55,
+        "functions": 65,
+        "lines": 72,
+        "statements": 71
```

Comment:
Feeling iffy in regard to these thresholds, i understand having higher thresholds will probably lead to failed builds, I would suggest leaving out thresholds for now, we would of course require to add an issue to add them later.
@cKellyDesign thoughts?

Reply:
@p-netm @kelvin-muchiri - I'm ok with doing this piecemeal, and if we do then does this PR actually close #659, or would the followup PR? Maybe if we leave the thresholds in we can close the ticket and open a new issue to remove them if builds fail (...unreasonably).

@moshthepitt thoughts?

Reply:
> @p-netm @kelvin-muchiri - I'm ok with doing this piecemeal, and if we do then does this PR actually close #659, or would the followup PR? Maybe if we leave the thresholds in we can close the ticket and open a new issue to remove them if builds start failing (...unreasonably).

Yea, the PR doesn't seem to close issue #659, at least not for the second task.

Reply:
@kelvin-muchiri AFAIK I think it would also close the second task,  all that would be needed is a proper integration between coveralls and github, such that if coveralls detected a drop in coverage it would then report that to github which would inturn autofail the build. I believe reaveal-fe's admins would also have to tweak some configuration for the repo, to make sure merge is disabled for prs that drop the coverage.